### PR TITLE
[DOC] Updated readme and contributing files

### PR DESCRIPTION
**Pull request by outoftardis**
# Description

Updates to readme/contributing files based on the request to unify Intel repos + small additional fixes.

Output:
- [README](https://github.com/outoftardis/daal/tree/doc-readme-contribution#intel-oneapi-data-analytics-library-)
- [CONTRIBUTING](https://github.com/outoftardis/daal/blob/doc-readme-contribution/CONTRIBUTING.md#how-to-contribute)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -230,5 +230,18 @@ This repository contains branches corresponding to both oneAPI and classical ver
...
+|oneDAL       |2021.1-beta10|[master](https://github.com/oneapi-src/oneDAL)</br>[rls/onedal-beta10-rls](https://github.com/oneapi-src/oneDAL/tree/rls/onedal-beta10-rls)|&nbsp;&nbsp;&nbsp;[Home page](https://software.intel.com/content/www/us/en/develop/tools/oneapi/components/onedal.html)&nbsp;&nbsp;&nbsp;</br>&nbsp;&nbsp;&nbsp;[Documentation](http://oneapi-src.github.io/oneDAL/)&nbsp;&nbsp;&nbsp;</br>&nbsp;&nbsp;&nbsp;[System Requirements](https://software.intel.com/content/www/us/en/develop/articles/system-requirements-for-oneapi-data-analytics-library.html)|
+|Intel&reg; DAAL|2020 Update 3|[rls/daal-2020-u3-rls](https://github.com/oneapi-src/oneDAL/tree/rls/daal-2020-u2-rls)|&nbsp;&nbsp;&nbsp;[Home page](https://software.intel.com/content/www/us/en/develop/tools/data-analytics-acceleration-library.html)&nbsp;&nbsp;&nbsp;</br>&nbsp;&nbsp;&nbsp;[Developer Guide](https://software.intel.com/content/www/us/en/develop/documentation/daal-programming-guide/top.html)&nbsp;&nbsp;&nbsp;</br>&nbsp;&nbsp;&nbsp;[System Requirements](https://software.intel.com/content/www/us/en/develop/articles/intel-data-analytics-acceleration-library-2020-system-requirements.html)|
```
Comment (napetrov) [inline README.md:234]: I think link should be updated to u3 branch as well

Comment (outoftardis) [reply README.md:234]: yep, missed that. thanks!

**Comment by napetrov** [comment]
Have general comment on signing - is it required?  Can we say it's optional? 

Because most of our own commits are not signed and it's strange to request this from others.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -16,17 +16,30 @@
 *******************************************************************************/-->
 
 # How to Contribute
-We welcome community contributions to Intel(R) oneAPI Data Analytics Library. If you have an idea how to improve the product, you can:
+We welcome community contributions to Intel(R) oneAPI Data Analytics Library. You can:
 
-* Let us know about your proposal via [Issues on oneDAL GitHub\*](https://github.com/oneapi-src/oneDAL/issues).
-* Contribute your changes directly to the repository through [pull request](#pull-requests). 
+- Submit your changes directly with a [pull request](https://github.com/oneapi-src/oneDAL/pulls).
+- Log a bug or feedback with an [issue](https://github.com/oneapi-src/oneDAL/issues).
```
Comment (averbukh) [inline CONTRIBUTING.md:22]: feedback -> feature request ?

It's a little bit confusing to request feedbacks as an 'issue'

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -16,17 +16,30 @@
...
+
+Refer to our guidelines on [pull requests](#pull-requests) and [isssues](#issues) before you proceed.
```
Comment (averbukh) [inline CONTRIBUTING.md:24]: [isssues] -> [issues]

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -16,17 +16,30 @@
...
+
+## Issues
+
+Use [GitHub issues](https://github.com/oneapi-src/oneDAL/issues) to:
+- report an issue
+- provide feedback
```
Comment (averbukh) [inline CONTRIBUTING.md:30]: I propose to use another channels to get feedbacks.

It's a little bit confusing to request feedbacks as an 'issue'

File: CONTRIBUTING.md (hunk 3 above)
Comment (outoftardis) [reply CONTRIBUTING.md:24]: fixed, thanks

File: CONTRIBUTING.md (hunk 4 above)
Comment (outoftardis) [reply CONTRIBUTING.md:30]: agree, fixed

File: CONTRIBUTING.md (hunk 2 above)
Comment (outoftardis) [reply CONTRIBUTING.md:22]: agree

**Comment by outoftardis** [comment]
> Have general comment on signing - is it required? Can we say it's optional?
> 
> Because most of our own commits are not signed and it's strange to request this from others.

Discussed offline, agreed to opt out of signing for now.

**Review by napetrov — APPROVED** [review]

**Review by averbukh — APPROVED** [review]

**Pull request merged**